src/main/scala/ru/chermenin/spark/sql/execution/streaming/state/RocksDbStateStoreProvider.scala

```diff
@@ -701,12 +726,50 @@ object RocksDbStateStoreProvider {
     cacheBuilderWithOptions.build[UnsafeRow, String](loader)
   }
 
-  private def setTTL(conf: Map[String, String]): Int =
+  /**
+    * Creates a mapping of Streaming Query and its expiry timeout (seconds).
+    * For backward compatibility, an additional entry is done for [[UNNAMED_QUERY]]'s
+    *
+    * The timeout value for [[UNNAMED_QUERY]]'s is set by the value of [[STATE_EXPIRY_SECS]]
+    *
+    * @param stateStoreConf state store config map set on [[SparkConf]]
+    * @return mapping of queryName -> expirySecs
+    */
+
+  private def getExpirationByQuery(stateStoreConf: Map[String, String]): Map[String, Int] =
+    stateStoreConf
+      .filter(_._1.startsWith(s"$STATE_EXPIRY_SECS."))
+      .map(
+        c => (c._1.replace(s"$STATE_EXPIRY_SECS.", ""), getTTL(c._2)
+        )
+      )
+      .+(
+        (
+          UNNAMED_QUERY,
+          stateStoreConf.getOrElse(STATE_EXPIRY_SECS, DEFAULT_STATE_EXPIRY_SECS).toInt
+        ) // For backward compatibility
+      )
 
-    Try(conf.getOrElse(STATE_EXPIRY_SECS, DEFAULT_STATE_EXPIRY_SECS).toInt) match {
-      case Success(value) => value
-      case Failure(e) => throw new IllegalArgumentException(e)
+  /**
+    * Helper method to check if the given string is an integer
+    *
+    * @param value [[String]] value to be checked
+    * @return [[Option]] if value is not an integer returns [[None]] else [[Some]]
+    */
+  private def isInt(value: String): Option[Int] = {
```

Comment:
Seems, this method name suggests returning a boolean value. It would be better to rename it to something like `toInt` or `toOptionInt`.